src/trrntzip.c

```diff
@@ -504,6 +494,22 @@ int MigrateZip(const char *zip_path, const char *pDir, WORKSPACE *ws,
   logprint(stdout, mig->fProcessLog, "Rezipping - %s\n", szZipFileName);
   logprint(stdout, mig->fProcessLog, "%s\n", DIVIDER);
 
+  // This use of mktemp() is inherently unsafe! Better use mkstemp()
+  // if available, or at least ensure the file is created with O_EXCL.
+  mktemp(szTmpZipFileName);
```

Comment:
I think we should check the return value here instead of `!szTmpZipFileName[0]`.

Reply:
Hm, on your system. On my (NetBSD) system [it says](https://man.netbsd.org/mktemp.1):
```
RETURN VALUES
     The mktemp() and mkdtemp() functions return a pointer to the template on
     success and NULL on failure.
```
and [POSIX](https://pubs.opengroup.org/onlinepubs/009695399/functions/mktemp.html) (which removed the function in the meantime) said:
```
The mktemp() function shall return the pointer template. If a unique name cannot be created, template shall point to a null string.
```
So we have three different behaviours in case of errors :(
We really should switch away from `mktemp`.

Reply:
It's just 2 behaviours since my system agrees with POSIX, but we still need to check the return value and the template. I assume that covers Windows behaviour, too.
Do you know whether `mkstemp()` is available on Windows? Using that would be much better.